backend/bitboxbase/updater/updater.go

```diff
@@ -83,7 +108,150 @@ func (updater *Updater) Connect(address string, bitboxBaseID string) error {
+		Pattern:       noise.HandshakeXX,
+		StaticKeypair: *keypair,
+		Prologue:      []byte("Noise_XX_25519_ChaChaPoly_SHA256"),
+		Initiator:     true,
+	})
+	if err != nil {
+		return errors.New("failed to generate a new noise handshake state for the wallet app communication with the BitBox Base")
+	}
+	//Check the websocket connection
+	err = client.WriteMessage(1, []byte(opICanHasHandShaek))
+	if err != nil {
+		return errors.New("unable to write BitBox Base Handshake request to websocket")
+	}
+	_, responseBytes, err := client.ReadMessage()
+	responseSuccess := []byte("ACK")
+	if err != nil {
+		return errors.New("unable to read BitBox Base Handshake response from websocket")
+	}
+	if string(responseBytes) != string(responseSuccess) {
+		return errors.New("no ACK received from BitBox Base as response to hanshake request")
+	}
+
+	// Do 3 part noise 'XX' handshake.
+	msg, _, _, err := handshake.WriteMessage(nil, nil)
+	if err != nil {
+		return errors.New("noise failed to write the first handshake message")
+	}
+	err = client.WriteMessage(websocket.TextMessage, msg)
+	if err != nil {
+		return errors.New("the websocket failed writing the first noise handshake message")
+	}
+	_, responseBytes, err = client.ReadMessage()
+	if err != nil {
+		return errors.New("the websocket failed reading the second noise handshake message")
+	}
+	_, _, _, err = handshake.ReadMessage(nil, responseBytes)
+	if err != nil {
+		return errors.New("noise failed to read the second handshake message")
+	}
+	msg, updater.receiveCipher, updater.sendCipher, err = handshake.WriteMessage(nil, nil)
+	if err != nil {
+		return errors.New("noise failed to write the third handshake message")
+	}
+	err = client.WriteMessage(websocket.TextMessage, msg)
+	if err != nil {
+		return errors.New("the websocket failed writing the third handshake message")
+	}
+
+	// Check if the user already authenticated the channel binding hash
+	updater.baseNoiseStaticPubkey = handshake.PeerStatic()
+	pairingVerificationRequiredByApp := !updater.configContainsBitBoxBaseStaticPubkey(
+		updater.baseNoiseStaticPubkey)
+	if pairingVerificationRequiredByApp {
+		msg = []byte(opICanHasPairinVerificashun)
+	}
+	err = client.WriteMessage(websocket.TextMessage, updater.sendCipher.Encrypt(nil, nil, msg))
+	if err != nil {
+		return errors.New("the websocket failed writing the pairingVerificationRequiredByApp message at the verification stage of the noise handshake")
+	}
+	_, responseBytes, err = client.ReadMessage()
+	if err != nil {
+		return errors.New("the websocket failed writing the pairingVerificationRequiredByBitBoxBase message at the verification stage of the noise handshake")
+	}
+	responseBytes, err = updater.receiveCipher.Decrypt(nil, nil, responseBytes)
+	if err != nil {
+		return errors.New("noise failed decrypting the pairingVerificationRequredByBitBoxBase response bytes at the verification stage of the noise handshake")
+	}
+	pairingVerificationRequiredByBase := false
```

Comment:
those lines are the same as (a bit shorter):

`pairingVerificationRequiredByBase :=  string(responseBytes) == opICanHasPairinVerificashu`